### PR TITLE
persist-txn: do incremental reads in the maelstrom test

### PR DESCRIPTION
**Pull request by danhhz**
This both exercises more edge cases as well as more directly models the OCC protocol we're planning to use for Pv2 Use Case Isolation. We restart all of them at the beginning of each request, so that we continue to test starting a new read reasonably frequently.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by danhhz** [comment]
TFTR!

File: src/persist-cli/src/maelstrom/txn_list_append_multi.rs [hunk 1]
```diff
@@ -310,43 +325,6 @@ impl Transactor {
         self.read_at(read_ts, data_ids).await
     }
 
-    fn read_data_at(
```
Comment (aljoscha) [inline src/persist-cli/src/maelstrom/txn_list_append_multi.rs:313]: 🙌

**Review by aljoscha — APPROVED** [review]
I had these comments/questions lying around 🙈

File: src/persist-cli/src/maelstrom/txn_list_append_multi.rs [hunk 2]
```diff
@@ -488,7 +466,7 @@ impl Service for TransactorService {
                             cfg,
                             "maelstrom".to_owned(),
                             mz_repr::Timestamp::minimum(),
-                            SYSTEM_TIME.clone(),
+                            NOW_ZERO.clone(),
```
[the remark this answers is not in the document]
Comment (danhhz) [reply src/persist-cli/src/maelstrom/txn_list_append_multi.rs:469]: Correct, not needed to make it work! It was an attempt to make the timestamps be smaller numbers, must easier to scan them and see the difference between e.g. 7 and 207 than between two epoch millis. It actually didn't work, we're still getting epoch millis, but I didn't dig into why and forgot to pull it out

**Pull request merged**